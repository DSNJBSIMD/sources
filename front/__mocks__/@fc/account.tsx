import type { PropsWithChildren } from 'react';
import React from 'react';

export const AccountContext = React.createContext(expect.any(Object));

export const AccountProvider = jest.fn(
  ({ children, config }: { config: unknown } & PropsWithChildren) => (
    <AccountContext.Provider value={config}>
      <div data-mockid="AccountProvider">
        <div>{children}</div>
      </div>
    </AccountContext.Provider>
  ),
);

export const ConnectValidator = {
  validate: jest.fn(),
};

export const Options = {
  CONFIG_NAME: 'Account',
};
