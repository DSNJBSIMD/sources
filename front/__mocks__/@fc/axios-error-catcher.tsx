import type { PropsWithChildren } from 'react';
import React from 'react';

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));

export const AxiosErrorCatcherProvider = jest.fn(
  ({ children, value }: { value: unknown } & PropsWithChildren) => (
    <AxiosErrorCatcherContext.Provider value={value}>{children}</AxiosErrorCatcherContext.Provider>
  ),
);
