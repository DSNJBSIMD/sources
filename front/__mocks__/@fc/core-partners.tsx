export const CreateInstanceButton = jest.fn(() => <div data-mockid="CreateInstanceButton" />);

export const InstanceComponent = jest.fn(() => <div data-mockid="InstanceComponent" />);

export const InstancesListComponent = jest.fn(() => <div data-mockid="InstancesListComponent" />);

export const PartnersService = {
  commit: jest.fn(),
  get: jest.fn(),
};

export enum Environment {
  SANDBOX = 'SANDBOX',
  PRODUCTION = 'PRODUCTION',
}

export enum Options {
  CONFIG_NAME = 'Partners',
}

export enum PublicationStatus {
  DRAFT = 'DRAFT',
  PENDING = 'PENDING',
  PUBLISHED = 'PUBLISHED',
}
