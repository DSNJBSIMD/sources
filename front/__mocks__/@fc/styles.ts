import type { ReactElement } from 'react';
import React from 'react';

import type { StylesProviderProps } from '@fc/styles';

export const useStylesQuery = jest.fn();

export const useStylesContext = jest.fn();

export const useStylesVariables = jest.fn(() => []);

export const StylesProvider = jest.fn(
  ({ children }: StylesProviderProps) => children as ReactElement,
);

export const StylesContext = React.createContext(expect.any(Object));

export enum Strings {
  DOUBLE_DASH = '--',
}
