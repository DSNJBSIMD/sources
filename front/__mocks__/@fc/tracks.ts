export const TracksListComponent = jest.fn();

export enum Options {
  CONFIG_NAME = 'Tracks',
}
