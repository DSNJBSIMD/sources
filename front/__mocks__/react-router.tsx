const actualModule = jest.requireActual('react-router');

module.exports = {
  ...actualModule,
  BrowserRouter: jest.fn(({ children, element }) => {
    if (element) {
      return element;
    }
    return <div data-mockid="BrowserRouter">{children}</div>;
  }),
  Link: jest.fn(({ children }) => <span data-mockid="Link">{children}</span>),
  NavLink: jest.fn(({ children }) => <div data-mockid="NavLink">{children}</div>),
  Navigate: jest.fn(() => <div>ReactRouterDom Navigate Component</div>),
  Outlet: jest.fn(() => <div>ReactRouterDom Outlet Component</div>),
  Route: jest.fn(({ children, element }) => {
    if (element) {
      return element;
    }
    return <div data-mockid="Route">{children}</div>;
  }),
  Routes: jest.fn(({ children }) => children),
  ScrollRestoration: jest.fn(() => <span data-mockid="ScrollRestoration" />),
  isRouteErrorResponse: jest.fn(),
  matchPath: jest.fn(),
  redirect: jest.fn(),
  useLoaderData: jest.fn(),
  useLocation: jest.fn(() => ({
    pathname: '',
  })),
  useNavigate: jest.fn(() => jest.fn()),
  useOutletContext: jest.fn(),
  useParams: jest.fn(),
  useRouteError: jest.fn(),
  useRouteLoaderData: jest.fn(),
  useSearchParams: jest.fn(() => [new URLSearchParams(), jest.fn()]),
};

export {};
