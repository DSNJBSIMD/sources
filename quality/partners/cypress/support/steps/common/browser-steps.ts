import { When } from '@badeball/cypress-cucumber-preprocessor';

When(/^je rafraÃ®chis la page$/, function () {
  // cy.reload() is not working as a cy.visit
  cy.url().then((url) => cy.visit(url, { failOnStatusCode: false }));
  // Wait for the page to load
  // eslint-disable-next-line cypress/no-unnecessary-waiting
  cy.wait(500);
});
