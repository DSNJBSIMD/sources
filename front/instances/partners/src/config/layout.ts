import { LogoFranceConnect } from '@fc/assets';
import type { LayoutConfig } from '@fc/layout';

export const Layout: LayoutConfig = {
  features: {
    showLicence: false,
    showServiceTitle: true,
  },
  footer: {
    description:
      'L’Espace partenaires FranceConnect est un dispositif qui permet aux partenaires de gérer leur fournisseur de service FranceConnect.',
    links: [
      {
        external: true,
        href: 'https://www.legifrance.gouv.fr',
        label: 'legifrance.gouv.fr',
        title: 'Accéder au site legifrance.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.info.gouv.fr',
        label: 'info.gouv.fr',
        title: 'Accéder au site info.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.service-public.fr/',
        label: 'service-public.fr',
        title: 'Accéder au site service-public.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://data.gouv.fr',
        label: 'data.gouv.fr',
        title: 'Accéder au site data.gouv.fr nouvelle fenêtre',
      },
    ],
    navigation: [
      {
        external: false,
        href: '/plan-du-site',
        label: 'Plan du site',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/accessibilite',
        label: 'Accessibilité : non conforme',
      },
      {
        external: false,
        href: '/mentions-legales',
        label: 'Mentions légales',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Données personnelles',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Gestion des cookies',
      },
    ],
  },
  service: {
    baseline: 'Espace Partenaires',
    homepage: '/',
    logo: LogoFranceConnect,
    name: 'FranceConnect',
  },
  sitemap: [
    {
      href: '/',
      label: 'Accueil',
    },
    {
      href: '/mentions-legales',
      label: 'Mentions Légales',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/licence',
      label: 'Licence',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/accessibilite',
      label: 'Accessibilité',
    },
  ],
};
