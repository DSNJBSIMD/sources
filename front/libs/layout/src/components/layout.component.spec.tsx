import { render } from '@testing-library/react';
import { Outlet, ScrollRestoration } from 'react-router';

import { useSafeContext } from '@fc/common';

import { LayoutFooterComponent } from './footer';
import { LayoutHeaderComponent } from './header';
import { ApplicationLayout } from './layout.component';

// Given
jest.mock('./footer/layout-footer.component');
jest.mock('./header/layout-header.component');

describe('ApplicationLayout', () => {
  beforeEach(() => {
    // Given
    jest.mocked(useSafeContext).mockReturnValue(expect.any(Object));
    // @NOTE by implementation JEST.DOM render component into a <div />
    // <html element cannot be a child of a <div /> element
    // we cannot use JEST.DOM.render(..., { container }) option, because
    // there's no parent container for a html element
    jest.spyOn(console, 'error').mockImplementation(jest.fn());
  });

  it('should match snapshot', () => {
    // When
    const { container } = render(<ApplicationLayout />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render ScrollRestoration', () => {
    // When
    render(<ApplicationLayout />);

    // Then
    expect(ScrollRestoration).toHaveBeenCalledOnce();
    expect(ScrollRestoration).toHaveBeenCalledWith({}, undefined);
  });

  it('should render LayoutHeaderComponent without props', () => {
    // When
    render(<ApplicationLayout />);

    // Then
    expect(LayoutHeaderComponent).toHaveBeenCalledOnce();
    expect(LayoutHeaderComponent).toHaveBeenCalledWith({}, undefined);
  });

  it('should render LayoutFooterComponent without props', () => {
    // When
    render(<ApplicationLayout />);

    // Then
    expect(LayoutFooterComponent).toHaveBeenCalledOnce();
    expect(LayoutHeaderComponent).toHaveBeenCalledWith({}, undefined);
  });

  it('should render Outlet without props', () => {
    // When
    render(<ApplicationLayout />);

    // Then
    expect(Outlet).toHaveBeenCalledOnce();
    expect(Outlet).toHaveBeenCalledWith({}, undefined);
  });
});
