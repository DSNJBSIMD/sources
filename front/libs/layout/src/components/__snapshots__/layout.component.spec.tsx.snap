// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationLayout should match snapshot 1`] = `
<div>
  <span
    data-mockid="ScrollRestoration"
  />
  <div
    data-mockid="Helmet"
  >
    <span
      data-htmltag="html"
    />
  </div>
  <div
    class="sticky-body"
  >
    <div
      class="sticky-content"
    >
      <div>
        LayoutHeaderComponent
      </div>
      <div>
        ReactRouterDom Outlet Component
      </div>
    </div>
    <div>
      LayoutFooterComponent
    </div>
  </div>
</div>
`;
